spec/models/transformation_mapping_spec.rb

```diff
@@ -1,37 +1,67 @@
 RSpec.describe TransformationMapping, :v2v do
-  let(:src_ems) { FactoryBot.create(:ems_vmware) }
-  let(:dst_ems_redhat) { FactoryBot.create(:ems_redhat) }
+  let(:src_ems_vmware)    { FactoryBot.create(:ems_vmware) }
+  let(:dst_ems_redhat)    { FactoryBot.create(:ems_redhat) }
   let(:dst_ems_openstack) { FactoryBot.create(:ems_openstack) }
-  let(:src_cluster) { FactoryBot.create(:ems_cluster, :ext_management_system => src_ems) }
+
+  let(:src_cluster_vmware) { FactoryBot.create(:ems_cluster, :ext_management_system => src_ems_vmware) }
   let(:dst_cluster_redhat) { FactoryBot.create(:ems_cluster, :ext_management_system => dst_ems_redhat) }
+
+  let(:src_hosts_vmware) { FactoryBot.create_list(:host_vmware, 1, :ems_cluster => src_cluster_vmware) }
+  let(:dst_hosts_redhat) { FactoryBot.create_list(:host_redhat, 1, :ems_cluster => dst_cluster_redhat) }
+
+  let(:src_storages_vmware) { FactoryBot.create_list(:storage, 1, :hosts => src_hosts_vmware) }
+  let(:dst_storages_redhat) { FactoryBot.create_list(:storage, 1, :hosts => dst_hosts_redhat) }
+
+  let(:src_lan_vmware) { FactoryBot.create(:lan) }
+  let(:dst_lan_redhat) { FactoryBot.create(:lan) }
+
   let(:dst_cloud_tenant_openstack) { FactoryBot.create(:cloud_tenant, :ext_management_system => dst_ems_openstack) }
-  let(:vm) { FactoryBot.create(:vm_vmware, :ems_cluster => src_cluster) }
 
   let(:mapping_redhat) do
-    FactoryBot.create(
-      :transformation_mapping,
-      :transformation_mapping_items => [TransformationMappingItem.new(:source => src_cluster, :destination => dst_cluster_redhat)]
-    )
+    FactoryBot.create(:transformation_mapping).tap do |tm|
+      tm.transformation_mapping_items = [
+        FactoryBot.create(:transformation_mapping_item,
+          :source => src_cluster_vmware,
+          :destination => dst_cluster_redhat,
+          :transformation_mapping => tm
```

Comment:
I don't think you need to do both `tm.transformation_mapping_items` and `:transformation_mapping => tv`.
Actually without doing `tm.save!` at the end I think `tm.transformation_mapping_items = [` isn't doing anything.

Reply:
I didn't think it would be required either, but if you put a validation in `TransformationMappingItem` it will bomb because the relationship still isn't set yet. I don't think a .save is required, otherwise the specs would fail.